lib/mix/lib/mix/tasks/test.ex

```diff
@@ -679,14 +749,66 @@ defmodule Mix.Tasks.Test do
     end
   end
 
-  defp display_warn_test_pattern(test_files, test_pattern, matched_test_files, warn_test_pattern) do
-    files = Mix.Utils.extract_files(test_files, warn_test_pattern) -- matched_test_files
+  defp classify_test_files(potential_test_files, project) do
+    test_load_filters = project[:test_load_filters] || [&String.ends_with?(&1, "_test.exs")]
+    elixirc_paths = project[:elixirc_paths] || []
 
-    for file <- files do
-      Mix.shell().info(
-        "warning: #{file} does not match #{inspect(test_pattern)} and won't be loaded"
-      )
-    end
+    # ignore any _helper.exs files and files that are compiled (test support files)
+    test_ignore_filters =
+      Keyword.get_lazy(project, :test_ignore_filters, fn ->
+        [
+          &String.ends_with?(&1, "_helper.exs"),
+          fn file -> Enum.any?(elixirc_paths, &String.starts_with?(file, &1)) end
```

Comment:
I think we should include this filter by default, because if the user wants to add a new one, we should not ask them to reimplement this. WDYT? I am not sure if we should exclude the test_helper.exs by default too, but we probably should. So the user simply passes additional filters?

Reply:
Changed! If anyone feels the need to change those defaults, they'll hopefully open up an issue :)